### DISCUSSION TITLE
Data set input validation

### DISCUSSION THREAD

**Discussion by katelynienaber** [zowe/zowe-explorer-vscode#1141] [Ideas]
This is related to [this PR](https://github.com/zowe/vscode-extension-for-zowe/pull/1092) and also to template creation.

I'm working on allowing users to make their own DS templates. To do that I want to disable the standard templates once someone has created 1 custom template (with the option to turn them back on in settings.json)

Aaand of course there are some fields that users can't leave blank when creating a DS, without the standard template. So branch [ds-attribute-validation](https://github.com/zowe/vscode-extension-for-zowe/tree/ds-attribute-validation) creates a framework for validating user input in the DS attributes before allocating.

This discussion is open for people to add their own validation suggestions. So that we can cover the most important ones and skip things that shouldn't be dealt with in Explorer :)

**Comment by jellypuno** [comment]
@katelynienaber This is just for validation of DS Attributes and not the Dataset Name, right?

**Reply by katelynienaber** [reply]
No, it's for both. DS name is treated as an attribute up until allocation with the Imperative function. So we can do validation of attributes AND DS name in this branch.

**Reply by jellypuno** [reply]
ok. so there is an on-going update from z/OSMF regarding DSName validation. And it is regarding codespaces. With this update, the validation for the DSName will not go to Zowe Explorer but through the API itself. @dkelosky knows more but I would suggest to leave the DSName validation out. Unless..... the validation is just for the total number of characters and the rule that a number is not allowed after a period.

For the other attributes, are you going to validate them using the IBM syntax? https://www.ibm.com/support/knowledgecenter/en/SSLTBW_2.3.0/com.ibm.zos.v2r3.idan400/dacre.htm

**Reply by katelynienaber** [reply]
Yes, I just wasn't sure if I should validate to the fullest extent of the law XD

**Reply by dkelosky** [reply]
we used to have data set name validation within zowe CLI but it failed pretty fast when we had to handle various encodings (we ended up disallowing creating valid data set names).  so, our stance has been to rely on server validation and surface those errors instead.

**Reply by jellypuno** [reply]
> Yes, I just wasn't sure if I should validate to the fullest extent of the law XD

Well, if you are going to implement half of it, the users will start  looking for the other half. :D So I do agree with Dan. We should rely on server validation and surface those errors.

**Reply by katelynienaber** [reply]
@dkelosky @jellypuno If we will just check if fields are not blank...which fields should be on the list?

Jelly was suggesting these, which I put into this branch:
- SPACE (the CYL(primary, secondary)
- Record Format (RECFM = FB, VB etc)
- Record length (LRECL = 80)
- Disposition (DISP=SHR)

Anything else? Any weird edge cases where something extra should be added/left out of the list?
